comp/core/tagger/taggerimpl/tagstore/entity_tags.go

```diff
@@ -147,3 +200,160 @@ func (e *EntityTags) shouldRemove() bool {
 
 	return true
 }
+
+func (e *EntityTagsWithMultipleSources) tagsForSource(source string) *sourceTags {
+	tags, ok := e.sourceTags[source]
+	if !ok {
+		return nil
+	}
+	return &tags
+}
+
+func (e *EntityTagsWithMultipleSources) setTagsForSource(source string, tags sourceTags) {
+	e.sourceTags[source] = tags
+	e.cacheValid = false
+}
+
+func (e *EntityTagsWithMultipleSources) tagsBySource() map[string][]string {
+	tagsBySource := make(map[string][]string)
+
+	for source, tags := range e.sourceTags {
+		allTags := append([]string{}, tags.lowCardTags...)
+		allTags = append(allTags, tags.orchestratorCardTags...)
+		allTags = append(allTags, tags.highCardTags...)
+		tagsBySource[source] = allTags
+	}
+
+	return tagsBySource
+}
+
+func (e *EntityTagsWithMultipleSources) sources() []string {
+	sources := make([]string, 0, len(e.sourceTags))
+	for source := range e.sourceTags {
+		sources = append(sources, source)
+	}
+	return sources
+}
+
+func (e *EntityTagsWithMultipleSources) deleteSource(source string, expiryDate time.Time) {
+	tags, ok := e.sourceTags[source]
+	if !ok {
+		return
+	}
+
+	tags.expiryDate = expiryDate
+	e.sourceTags[source] = tags
+}
```

Comment:
Some questions about this method:
- Why is the name `deleteSource` while it doesn't actually delete the source, it only sets the expiration date for the source. Do you think `setSourceExpiration` is more relevant to what the method does?
- why do we need to run the last line of this method ( `e.sourceTags[source] = tags`)? It seems like we are setting `e.sourceTags[source]` to itself. Am I missing something?
-

Reply:
I renamed the function in the new commit :+1: